### PR TITLE
Fix structure packing in DiskCryptor stuff

### PR DESCRIPTION
**Pull request by ivan-freed**
This might fix https://github.com/magnumripper/JohnTheRipper/issues/3542 and address https://github.com/magnumripper/JohnTheRipper/pull/3540 as well.

### CODE REVIEW

**Comment by ivan-freed** [comment]
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=52991 could be relevant for us.

File: src/diskcryptor_common.h [hunk 1]
```diff
@@ -27,6 +27,17 @@ struct custom_salt {
...
+#ifdef __GNUC__
+#define PACKED __attribute__ ((__packed__))
+#else
+#define PACKED
+#endif
+
+#ifndef __GNUC__
+#define PACKED
+#pragma pack(push,1)
+#endif
```
Comment (magnumripper) [inline src/diskcryptor_common.h]: Perhaps move the lower one of these to the else clause of the upper one?

Comment (ivan-freed) [reply src/diskcryptor_common.h]: I am not about this but perhaps the `#pragma pack(pop)` part needs to come after the structure definition (and thus needs to belong to a new `ifdef` block)?

**Comment by ivan-freed** [comment]
@jfoug Does this PR help with getting things built in Visual C?

**Pull request merged**